packages/core/lib/commands/test/run.js

```diff
@@ -7,32 +7,14 @@ const parseCommandLineFlags = options => {
   const reporter = options.reporter || options.r;
 
   /**
-   * This if-else condition is explicitly written to avoid the overlapping of
-   * the config by the default mocha reporter type when user specifies a mocha reporter type
-   * in the config and doesn't specify it as the command line argument.
-   * If the reporter is returned as undefined, it ignores the specification of any reporter type in the
-   * config and displays the default mocha reporter "spec", as opposed to reporter completely being absent
-   * which results in checking for the reporter type specified in the config.
+   * Reporter value passed on the command line has precedence over a reporter value in the config.
```

Comment:
I agree that the earlier coment was overlong, but the way that you simplified it erased the key reason that the comment was necessary -- namely, that in this case the distinction between `reporter` being undefined, and `reporter` being absent, **is relevant**, and that the code must not be simplified to remove this distinction.  IMO, the comment should still indicate that (it just doesn't need to take so many words to do it).

Reply:
Thanks for pointing that out @haltman-at . I think it is worth mentioning. I will add that before merging.

Reply:
@haltman-at I, have added the previous comment which was left out. I thought of making the comment a bit shorter but I think these many lines in the comment are important to completely understand the logic.

Reply:
Hm, I don't think the addition does the job.  What's non-obvious here isn't the **logic**.  The logic is clear!  What's non-obvious here is the **reason** for the logic.  Simply put: If a refactorer were to see this code, they might incorrectly attempt to simplify it by replacing it with `return { mocha: { grep, bail, reporter }}`, which wouldn't work!  They would do this because they would expect `undefined`, and complete absence, to act exactly the same, because they do so in many contexts.  But they don't here.  The key thing to get at is that this distinction, which is normally irrelevant, is in this case relevant.  I'd suggest a comment like "Note: It is important that `reporter` be completely omitted, and not be set to `undefined`!"  That's the sort of thing that would express that the distinction is important and that you not remove it.  (It might also be nice to say why -- i.e., that it being present as `undefined` will take precedence as over the default -- but the most important thing is to express the distinction and it's importance.)

Reply:
Okay, I got your point now. I will add a note like you suggested. Thanks for the suggestion @haltman-at . Sometimes it is difficult to convey the words in a proper way :)